actions/databases.go

```diff
@@ -0,0 +1,290 @@
+				if clusterNotFound {
+					return c.Error(404, aerr)
+				}
+			} else {
+				log.Println(aerr.Error())
+				return c.Error(400, aerr)
+			}
+		} else {
+			log.Println(err.Error())
+			return err
+		}
+	}
+
+	output := struct {
+		*rds.DescribeDBClustersOutput
+		*rds.DescribeDBInstancesOutput
+	}{
+		clustersOutput,
+		instancesOutput,
+	}
+	return c.Render(200, r.JSON(output))
+}
+
+// DatabasesPost creates a database in a given account
+//   It will create a database instance as specified by the `Instance` hash parameters.
+//   If a `Cluster` hash is also given, it will first create an RDS cluster and the instance next.
+func DatabasesPost(c buffalo.Context) error {
+	type DatabaseCreateInput struct {
+		// https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#CreateDBClusterInput
+		Cluster *rds.CreateDBClusterInput
+		// https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#CreateDBInstanceInput
+		Instance *rds.CreateDBInstanceInput
+	}
+	input := DatabaseCreateInput{}
+	if err := c.Bind(&input); err != nil {
+		log.Println(err)
+		return c.Error(400, err)
+	}
+	if input.Instance == nil {
+		return c.Error(400, errors.New("Bad request"))
+	}
+
+	rdsClient := RDS[c.Param("account")]
+
+	// create rds cluster first, if specified
+	if input.Cluster != nil {
+		if input.Cluster.DBSubnetGroupName == nil {
+			input.Cluster.DBSubnetGroupName = aws.String(rdsClient.DefaultSubnet)
+		}
+
+		clusterOutput, err := rdsClient.Service.CreateDBCluster(input.Cluster)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				log.Println(aerr.Error())
+				return c.Error(400, aerr)
+			}
+			log.Println(err.Error())
+			return err
+		}
+		log.Println("Created RDS cluster", clusterOutput)
+	}
+
+	// create rds instance
+	if input.Instance.DBSubnetGroupName == nil {
+		input.Instance.DBSubnetGroupName = aws.String(rdsClient.DefaultSubnet)
+	}
+
+	instanceOutput, err := rdsClient.Service.CreateDBInstance(input.Instance)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			log.Println(aerr.Error())
+			return c.Error(400, aerr)
+		}
+		log.Println(err.Error())
+		return err
+	}
+	log.Println("Created RDS instance", instanceOutput)
+
+	return c.Render(200, r.JSON(instanceOutput))
+}
+
+// DatabasesDelete deletes a database in a given account
+//   It will delete the database instance with the given {db} name and will also delete the associated cluster
+//   if the instance belongs to a cluster and is the last remaining member.
+//   If the snapshot=true parameter is given, it will create a final snapshot of the instance/cluster.
+func DatabasesDelete(c buffalo.Context) error {
+	// if snapshot param is given, a final snapshot will be created before deleting
+	snapshot := false
+	if b, err := strconv.ParseBool(c.Param("snapshot")); err == nil {
+		snapshot = b
+	}
+
+	rdsClient := RDS[c.Param("account")]
+	var instanceInput *rds.DeleteDBInstanceInput
+	var clusterInput *rds.DeleteDBClusterInput
```

Comment:
If you allocate these here instead of just defining them, you can build them up instead of repeating all of the properties down below for each conditional.